### DISCUSSION TITLE
Running an Elixir Kernel with Executable books

### DISCUSSION THREAD

**Discussion by J0** [executablebooks/meta#180] [Q&A] [answered]
Hey all!

I'm trying to run elixir kernels with [Thebe Lab](https://github.com/executablebooks/thebe) so that I can create a site with interactive elixir exercises. To achieve this goal I took the minimal starter from the documents and changed the `binderOptions`:
```
<body>

<script type="text/x-thebe-config">
    {
      requestKernel: true,
      binderOptions: {
        name: "Elixir",
        repo: "J0/binder-el",
        kernelName: "ielixir",
        path: "./.local/share/jupyter/kernels/ielixir",
      },
    }
  </script>
<script src="https://unpkg.com/thebelab@latest/lib/index.js"></script>

<button id="activateButton"  style="width: 150px; height: 75px; font-size: 1.5em;">Activate</button>
<script>
var bootstrapThebe = function() {
    thebelab.bootstrap();
}

document.querySelector("#activateButton").addEventListener('click', bootstrapThebe)
</script>
<pre data-executable="true" >IO.puts("Hello world")</pre>

</body>
```
The repository [J0/binder-el](https://github.com/J0/binder-el) is a repository with a single Dockerfile containing the setup required to run an Elixir kernel as per the [ielixir repository](https://github.com/pprzetacznik/IElixir). I have tested it before hand when using `binder.org` and I am able to successfully execute `IO.puts("Hello world")` in the notebook that I create.

However, when I attempt to run the example snippet above it seems to still be running an ipython kernel despite the fact that I have specified `kernelName: "ielixir"` which has left me slightly puzzled.
![image](https://user-images.githubusercontent.com/8011761/99906161-7de7c500-2d10-11eb-8b01-93f744830d36.png)

I'm just wondering if I am missing something or if anyone has run such a configuration before. Would deeply appreciate any suggestions as to how I might modify the  existing setup to support the ielixir kernel. 

Thank you!

**Comment by J0** [comment, marked as answer]
Resolved, see  https://stackoverflow.com/questions/64955349/how-might-we-use-thebelab-to-execute-a-code-cell-which-is-backed-by-a-jupyter-no

Thanks for building this project!

**Reply by chrisjsewell** [reply]
Thanks @J0! always good when someone answers their own question 😄

**Reply by choldgraf** [reply]
Indeed - and i look forward to your blog post, i suspect that it may help uncover some UX improvements to make in thebe
